tests/test_socket.py

```diff
@@ -228,3 +229,35 @@ class MySocket(socket.socket):
     """)
     result = testdir.runpytest("--verbose")
     assert_socket_blocked(result)
+
+
+def test_unix_domain_sockets_blocked_with_disable_socket(testdir):
+    testdir.makepyfile("""
+        import socket
+
+        def test_unix_socket():
+            socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+    """)
+    result = testdir.runpytest("--verbose", "--disable-socket")
+    if hasattr(socket, 'AF_UNIX'):
+        assert_socket_blocked(result)
+    else:
+        # no AF_UNIX on windows
+        result.assert_outcomes(passed=0, skipped=0, failed=1)
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**suggestion:** ‏I'm not a huge fan of conditional logic in tests, an improvement might be to use `@pytest.mark.skipif` to selectively skip these tests if the platform doesn't have unix sockets.
Right now the test is basically saying "I assume if you use AF_UNIX on Windows, the code fails" - and that's beyond the responsibility of this plugin, right?

Ref: https://docs.pytest.org/en/stable/reference.html#pytest-mark-skipif

If I'm reading this wrong, let me know!